go/vt/sqlparser/comments.go

```diff
@@ -376,6 +379,29 @@ func AllowScatterDirective(stmt Statement) bool {
 	return comments != nil && comments.Directives().IsSet(DirectiveAllowScatter)
 }
 
+// GetCriticalityFromStatement gets the criticality from the provided Statement, using DirectiveCriticality
+func GetCriticalityFromStatement(statement Statement) string {
+	commentedStatement, ok := statement.(Commented)
+	// This would mean that the statement lacks comments, so we can't obtain the workload from it. Hence default to
+	// empty criticality
+	if !ok {
+		return ""
+	}
+
+	directives := commentedStatement.GetParsedComments().Directives()
+	criticality, ok := directives.GetString(DirectiveCriticality, "")
+	if !ok || criticality == "" {
+		return ""
+	}
+
+	intCriticality, err := strconv.Atoi(criticality)
+	if err != nil || intCriticality < 0 || intCriticality > 100 {
+		return ""
```

Comment:
Invalid values are silently turned into empty string. We avoid this pattern. Please make this function return an error. If there is a criticality value, then we must ensure it is valid. Please use `vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, ...)` as the returned error.

Reply:
I can do that. To be clear, this will lead to a query with an invalid value being in a directive (which is a comment) being flat out failed, correct?

Reply:
that's correct. The directive has a meaning, and if the meaning is invalid, whatever code that has to deal with the directive has to fail. @harshit-gangal for double-checking.

Reply:
Fixed.